package/MDAnalysis/analysis/dssp/dssp.py

```diff
@@ -0,0 +1,293 @@
+"persistent", for clarity, and label them according to the structure
+that they maintain during the run:
+
+.. code-block:: python
+    from MDAnalysis.analysis.dssp import DSSP, translate
+    from MDAnalysisTests.datafiles import TPR, XTC
+    u = mda.Universe(TPR, XTC)
+    threshold = 0.8
+
+    long_run = DSSP(u).run()
+    persistent_residues = translate(
+        long_run
+        .results
+        .dssp_ndarray
+        .mean(axis=0) > threshold
+    )
+    print(''.join(persistent_residues)[:20])
+    # '--EEEE----------HHHH'
+
+
+Functions
+---------
+
+.. autofunction:: get_hbond_map
+.. autofunction:: assign
+.. autofunction:: translate
+
+Analysis classes
+----------------
+
+.. autoclass:: DSSP
+   :members:
+   :inherited-members:
+
+   .. attribute:: results.dssp
+
+       Contains the time series of the DSSP assignment
+       as chars N×m :class:`numpy.ndarray` array with content
+       ``[frame, residue]``, where `structure_type` is one of three characters:
+       ['H', 'E', '-'], representing alpha-helix, sheet and loop, respectively.
+
+   .. attribute:: results.dssp_ndarray
+
+       Contains the one-hot encoding of the time series of the DSSP assignment
+       as chars N×mx3 :class:`numpy.ndarray` array with content
+       ``[frame, residue, encoding]``, where `encoding` is a (3,) shape
+       :class:`numpy.ndarray` of booleans with axes representing loop '-',
+       helix 'H' and sheet 'E', consequtively.
+"""
+
+import numpy as np
+
+from ..base import AnalysisBase
+from ... import Universe
+
+try:
+    from pydssp.pydssp_numpy import (
+        assign,
+        get_hbond_map,
+        _get_hydrogen_atom_position,
+    )
+    HAS_PYDSSP = True
+
+except ModuleNotFoundError:
+    HAS_PYDSSP = False
+    from .pydssp_numpy import (
+        assign,
+        get_hbond_map,
+        _get_hydrogen_atom_position,
+    )
+
+
+def translate(onehot: np.ndarray) -> np.ndarray:
+    """Translate a one-hot encoding summary into char-based secondary
+    structure assignment. One-hot encoding corresponds to C3 notation:
+    '-', 'H', 'E' are loop, helix and sheet, respectively. Input array must
+    have its last axis of shape 3: (n_residues, 3) or (n_frames, n_residues, 3)
+
+    Examples
+    --------
+
+    >>> from MDAnalysis.analysis.dssp import translate
+    >>> import numpy as np
+    >>> # encoding 'HE-'
+    >>> onehot = np.array([
+        [False, True, False],  # 'H'
+        [False, False, True],  # 'E'
+        [True, False, False]]) # '-'
+    >>> ''.join(translate(onehot))
+    'HE-'
+
+    Parameters
+    ----------
+    onehot : np.ndarray
+        input array of one-hot encoding in ('-', 'H', 'E') order
+
+    Returns
+    -------
+    np.ndarray
+        array of '-', 'H' and 'E' symbols with secondary structure
+
+    .. versionadded:: 2.8.0
+    """
+    C3_ALPHABET = np.array(["-", "H", "E"])
+    index = np.argmax(onehot, axis=-1)
+    return C3_ALPHABET[index]
+
+
+class DSSP(AnalysisBase):
```

Comment:
Can you add a duecredit decorator for the Kabsch paper here?

Reply:
Can you point me to an example of its usage?

I tried searching `git grep -B 1 '^class'`, and also `git grep duecredit` and found nothing.

Reply:
Fixed.